### PR TITLE
s1335-url-enforcement

### PR DESCRIPTION
**Pull request by ozhey**

### CODE REVIEW

File: actions/get.action.yaml [hunk 1]
```diff
@@ -78,9 +78,4 @@ parameters:
       - "application/vnd.openxmlformats-officedocument.presentationml.presentation"
       - "application/msword"
       - "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
-      - "application/vnd.mozilla.xul+xml"
-  timeout:
-    type: "int"
-    description: "Timeout in seconds for http request"
-    default: 60
-    required: false
+      - "application/vnd.mozilla.xul+xml"
```
Comment (itsofirblink) [inline actions/get.action.yaml:81]: new line

File: actions/patch.action.yaml [hunk 2]
```diff
@@ -78,9 +78,4 @@ parameters:
...
       - "application/msword"
       - "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
-      - "application/vnd.mozilla.xul+xml"
-  timeout:
-    type: "int"
-    description: "Timeout in seconds for http request"
-    default: 60
-    required: false
\ No newline at end of file
+      - "application/vnd.mozilla.xul+xml"
```
Comment (itsofirblink) [inline actions/patch.action.yaml:81]: new line

File: implementation/http.go [hunk 3]
```diff
@@ -12,47 +12,84 @@ import (
...
+	h2, ok1 := credentials["Header 2"].(string)
+	v2, ok2 := credentials["Value 2"].(string)
+	if ok1 && ok2 {
+		req.Header.Add(h2, v2)
+	}
+	h3, ok1 := credentials["Header 3"].(string)
+	v3, ok2 := credentials["Value 3"].(string)
+	if ok1 && ok2 {
+		req.Header.Add(h3, v3)
 	}
```
Comment (itsofirblink) [inline implementation/http.go:92]: we support only 3 headers?
Create a function which iterates it :)

File: implementation/http.go [hunk 4]
```diff
@@ -127,30 +159,71 @@ func sendRequest(ctx *plugin.ActionContext, method string, urlAsString string, t
...
+	if !strings.HasPrefix(requestedURL.Host+requestedURL.Path, apiAddress.Host+apiAddress.Path) {
+		return errors.New("the requested url's host/path does not match the host/path defined in the connection. this is not allowed in order to prevent sending credentials to unwanted hosts/paths")
+	}
+	return nil
+}
+
+// fixURL adds https if the user forgot to include the scheme and trims unnecessary trailing slashes
+func fixURL(u *url.URL) error {
+	if u.Scheme == "" {
+		u.Scheme = "https"
```
Comment (itsofirblink) [inline implementation/http.go:219]: are we sure about it?

**Review by itsofirblink — CHANGES_REQUESTED** [review]

File: implementation/http.go (hunk 4 above)
Comment (ozhey) [reply implementation/http.go]: removed

File: implementation/http.go (hunk 3 above)
Comment (ozhey) [reply implementation/http.go]: Only 3 headers from the connection, yes. The user can as many headers as he requires in the action's parameters but we thought that in the connection itself 3 is enough, we don't want it to look too overloaded

File: actions/patch.action.yaml (hunk 2 above)
Comment (ozhey) [reply actions/patch.action.yaml]: done

File: actions/get.action.yaml (hunk 1 above)
Comment (ozhey) [reply actions/get.action.yaml]: done

**Review by itsofirblink — APPROVED** [review]

**Pull request merged**